src/io/flutter/sdk/FlutterSettingsConfigurable.java

```diff
@@ -142,24 +113,22 @@ public void apply() throws ConfigurationException {
 
   @Override
   public void reset() {
-    isModified = false;
-
     final FlutterSdk sdk = FlutterSdk.getGlobalFlutterSdk();
-    final String path = sdk != null ? FileUtilRt.toSystemDependentName(sdk.getHomePath()) : "";
+    final String path = sdk != null ? sdk.getHomePath() : "";
     sdkCombo.getComboBox().getEditor().setItem(path);
 
     updateVersionText();
     updateErrorLabel();
   }
 
   private void updateVersionText() {
-
     final FlutterSdk sdk = FlutterSdk.forPath(getSdkPathText());
     if (sdk == null) {
       versionDetails.setVisible(false);
     }
     else {
       try {
+        final ModalityState modalityState = ModalityState.current();
```

Comment:
Curious why you're hoisting this out of the runnable.  Don't we want the modality state when it gets invoked?